github/github_test.go

```diff
@@ -1157,8 +1164,8 @@ func TestDo_preservesResponseInHTTPError(t *testing.T) {
 	}
 
 	// Verify error type and access to status code
-	errResp, ok := err.(*ErrorResponse)
-	if !ok {
+	var errResp *ErrorResponse
+	if errors.As(err, &errResp) {
```

Comment:
Please change all usages of `errors.As` to `errors.Is` where the actual value is not needed to be cast to that type, like here any many other places:


```suggestion
	if errors.Is(err, ErrorResponse) {
```